### PR TITLE
[WIP] CMake: Link on ELF platforms with `-Bsymbolic-functions`

### PR DESCRIPTION
**Pull request by BertalanD**
By default, ELF platforms assume that all symbols defined in shared libraries may be interposed, so even locally defined, strongly-bound global functions will have to be called through the PLT.

TODO: Write the commit message :^)

MaskRay's blog post explains the historical and implementation details: https://maskray.me/blog/2021-05-16-elf-interposition-and-bsymbolic

Statistics with Lagom LibJS:
```
❯ bloaty libjs-after.so -- libjs-before.so
    FILE SIZE        VM SIZE
 --------------  --------------
  [ = ]       0  [DEL]      -8    [LOAD #3 [RW]]
 -89.2%    -592 -98.7%    -592    .plt.got
 -15.8% -1.10Ki -16.0% -1.10Ki    .got
 -38.8% -2.27Ki  [ = ]       0    [Unmapped]
  -0.9% -3.30Ki  -0.9% -3.30Ki    .rela.dyn
 -80.1% -12.1Ki -80.4% -12.1Ki    .got.plt
 -80.3% -24.3Ki -80.5% -24.3Ki    .plt
 -80.4% -36.4Ki -80.5% -36.4Ki    .rela.plt
  -0.4% -80.1Ki  -1.1% -77.9Ki    TOTAL
```

### CODE REVIEW

File: Userland/Libraries/LibUnicode/CMakeLists.txt [hunk 1]
```diff
@@ -14,4 +14,5 @@ set(GENERATED_SOURCES ${CURRENT_LIB_GENERATED})
 
 serenity_lib(LibUnicode unicode)
 target_link_libraries(LibUnicode PRIVATE LibCore)
+target_link_libraries(LibUnicode PRIVATE AllowInterposition)
```
Comment (trflynn89) [inline Userland/Libraries/LibUnicode/CMakeLists.txt:17]: You probably also want LibLocale - and I wonder how our weak definitions of the LibC regex functions deals with this (doesn't seem to fail tests?)

Comment (BertalanD) [reply Userland/Libraries/LibUnicode/CMakeLists.txt:17]: I'll try to help MaskRay's `-Bsymbolic-non-weak-functions` Binutils patch accepted + backported by us, which should do the right thing ™️

**Comment by nico** [comment]
This is nice, but at least for Ladybird, I'd expect that having a single statically-linked binary with `-fvisibility=hidden` will be even nicer. Maybe it's worth hacking up a prototype of that locally to get an idea of how much you're leaving on the table by not doing that?

**Comment by BertalanD** [comment]
We have recently had a discussion on Discord about shipping all deps as a single shared library for Ladybird and its helper processes. I think the conclusion was that we should still focus on ease/speed of development, which the current setup provides.

My main concern is currently the library loading overhead on Serenity CI, which this should still help with. Of course, that only goes so far, but let's get the easier.

The patch was easy to backport, and I can confirm that the system boots with it. Will try to roll it into the GCC 13.2 + Binutils 2.41 update PR.

**Pull request closed without merging**